rikaichan.js

```diff
@@ -164,15 +164,15 @@ var rcxMain = {
 				return false;
 			} */
 			try {
-				this.dict = new rcxDict(this.haveNames/* && !this.cfg.nadelay*/);
+				this.dict = new rcxDict();
+				return this.dict.init(this.haveNames/* && !this.cfg.nadelay*/);
 				//this.dict.setConfig(this.dconfig);
 			}
 			catch (ex) {
 				alert('Error loading dictionary: ' + ex);
-				return false;
 			}
 		}
-		return true;
+		return null;
```

Comment:
I think instead of returning null here we should pass true/false.

something like this above:

return this.dict.init(blahblah).then(() => {return false}, () => {alert('Error loading dictionary: ' + ex);})

and then here: return:
Promise.resolve(true);